### PR TITLE
brew-merge-homebrew: specify several merges in one command

### PR DESCRIPTION
**Pull request by maxim-belkin**
The idea is that with this change we should be able to use the following syntax:
```
brew merge-homebrew --core sha1 sha2 sha3
```
and it should create the following merge commits:
`origin -> sha1`, `sha1 -> sha2`, `sha2 -> sha3`

Further improvements will be to automatically split homebrew commits into patches based on which formulas are updated

### CODE REVIEW

**Comment by maxim-belkin** [comment]
Of course, specifying several commits will not work unless we automatically resolve merge commits :)

File: cmd/brew-merge-homebrew.rb [hunk 1]
```diff
@@ -26,15 +26,17 @@ def git_merge(fast_forward: false)
 
     safe_system git, "pull", "--ff-only", "origin", "master"
     safe_system git, "fetch", "homebrew"
-    start_sha1 = Utils.popen_read(git, "rev-parse", "origin/master").chomp
-    end_sha1 = Utils.popen_read(git, "rev-parse", "homebrew/master").chomp
+    homebrew_commits.each do |hbc|
```
Comment (sjackman) [inline cmd/brew-merge-homebrew.rb]: Can you please factor out `git_merge_commit(sha1, fast_forward: false)` so that this line becomes
```ruby
 homebrew_commits.each { |sha1| git_merge_commit sha1 }
```

File: cmd/brew-merge-homebrew.rb [hunk 2]
```diff
@@ -46,7 +48,7 @@ def resolve_conflicts
     safe_system HOMEBREW_BREW_FILE, "style", *conflicts
     safe_system git, "diff", "--check"
     safe_system git, "add", "--", *conflicts
-    conflicts
+    return conflicts
```
Comment (sjackman) [inline cmd/brew-merge-homebrew.rb]: Revert this change please. I think `brew style` complains about an unnecessary `return` statement.

File: cmd/brew-merge-homebrew.rb [hunk 3]
```diff
@@ -149,6 +151,15 @@ def merge_science
     puts "Now run:\n  git push homebrew && git push origin"
   end
 
+  def homebrew_commits
+    if ARGV.named.empty?
+      ["homebrew/master"]
+    else
+      ARGV.named.each { |hbc| safe_system git, "rev-parse", hbc }
```
Comment (sjackman) [inline cmd/brew-merge-homebrew.rb]: Add the option `--verify`

File: cmd/brew-merge-homebrew.rb (hunk 2 above)
Comment (maxim-belkin) [inline cmd/brew-merge-homebrew.rb]: I believe this causes
```
=> Conflicts
Formula/flex.rb Formula/gdk-pixbuf.rb Formula/libsoup.rb Formula/make.rb
Error: Failure while executing: Formula/flex.rb Formula/gdk-pixbuf.rb Formula/libsoup.rb Formula/make.rb
```

**Comment by maxim-belkin** [comment]
done!

**Comment by maxim-belkin** [comment]
@sjackman all comments addressed but please have a look at https://github.com/Linuxbrew/homebrew-developer/pull/31#discussion_r115615637

File: cmd/brew-merge-homebrew.rb [hunk 4]
```diff
@@ -19,22 +19,26 @@ def git
...
-    start_sha1 = Utils.popen_read(git, "rev-parse", "origin/master").chomp
-    end_sha1 = Utils.popen_read(git, "rev-parse", "homebrew/master").chomp
-
-    puts "Start commit: #{start_sha1}"
-    puts "End   commit: #{end_sha1}"
-
-    args = []
-    args << "--ff-only" if fast_forward
-    system git, "merge", *args, "homebrew/master", "-m", "Merge branch homebrew/master into linuxbrew/master"
+    homebrew_commits.each { |sha1| git_merge_commit sha1 }
```
Comment (sjackman) [inline cmd/brew-merge-homebrew.rb]: I got this wrong in my previous comment. It should be:
```ruby
homebrew_commits.each { |sha1| git_merge_commit sha1, fast_forward }
```

**Comment by sjackman** [comment]
> @sjackman all comments addressed but please have a look at https://github.com/Linuxbrew/homebrew-developer/pull/31#discussion_r115615637

Clicking this link takes me oddly to the top of this page, rather than a specific comment.

**Comment by maxim-belkin** [comment]
>> Revert this change please. I think brew style complains about an unnecessary return statement.

> I believe this causes
```ruby
=> Conflicts
Formula/flex.rb Formula/gdk-pixbuf.rb Formula/libsoup.rb Formula/make.rb
Error: Failure while executing: Formula/flex.rb Formula/gdk-pixbuf.rb Formula/libsoup.rb Formula/make.rb
```

**Comment by sjackman** [comment]
```ruby
  …
  conflicts
end
```
Should be :100:% equivalent to
```ruby
  …
  return conflicts
end
```

**Comment by sjackman** [comment]
Thanks, Maxim!
